src/mn_menu.c

```diff
@@ -3105,7 +3105,7 @@ void M_Drawer(void)
             }
 
             if ((patch_lump < 0
-                 || (W_IsIWADLump(patch_lump) && bigfont_lump >= 0))
+                 || (bigfont_lump >= 0 && (lumpinfo[patch_lump].wad_file != lumpinfo[bigfont_lump].wad_file)))
```

Comment:
I don't get it, why are we comparing `patch_lump' and `bigfont_lump'? `bigfont_lump` is always "DBIGFONT". I should refactor this.

Reply:
We are checking if they are from the same WAD file. If they are, i.e. if the same resource that provides us with the big font does also offer menu graphics, then we use the graphics.

Reply:
Oh, right, thank you.

Reply:
Problem is, if you load another PWAD providing menu graphics after the PWAD (or IWAD) providing the big font, the big font will still be used.

Reply:
> Problem is, if you load another PWAD providing menu graphics after the PWAD (or IWAD) providing the big font, the big font will still be used.

Comparing with `<` instead of `!=` might help, i.e. check if the big font providing WAD has been loaded *after* the menu graphics providing one.

Reply:
Regarding skill menu, now:

`-iwad doom.wad` uses Doom's menu graphics
`-iwad doom.wad -file chex3v.wad` uses Chex's menu graphics
`-iwad chex3v.wad` uses Chex's menu graphics
`-iwad chex3v.wad -file doom.wad` uses Doom's menu graphics

Reply:
I suggest just deleting these DBIGFONT checks. I made them to fix small inconsistencies in the Eviternity I-II menu, it's not worth it.

```diff
diff --git a/src/mn_menu.c b/src/mn_menu.c
index 38e13b9..bd4804b 100644
--- a/src/mn_menu.c
+++ b/src/mn_menu.c
@@ -3092,8 +3092,6 @@ void M_Drawer(void)
     // [FG] check current menu for missing menu graphics lumps - only once
     if (currentMenu->lumps_missing == 0)
     {
-        int bigfont_lump = W_CheckNumForName("DBIGFONT");
-
         for (int i = 0; i < max; i++)
         {
             const char *name = currentMenu->menuitems[i].name;
@@ -3104,9 +3102,7 @@ void M_Drawer(void)
                 patch_lump = W_CheckNumForName(name);
             }
 
-            if ((patch_lump < 0
-                 || (W_IsIWADLump(patch_lump) && bigfont_lump >= 0))
-                && currentMenu->menuitems[i].alttext)
+            if (patch_lump < 0 && currentMenu->menuitems[i].alttext)
             {
                 currentMenu->lumps_missing++;
                 break;
diff --git a/src/mn_setup.c b/src/mn_setup.c
index ffa5227..d1f106e 100644
--- a/src/mn_setup.c
+++ b/src/mn_setup.c
@@ -3722,9 +3722,8 @@ int MN_StringHeight(const char *string)
 void MN_DrawTitle(int x, int y, const char *patch, const char *alttext)
 {
     int patch_lump = W_CheckNumForName(patch);
-    int bigfont_lump = W_CheckNumForName("DBIGFONT");
 
-    if (patch_lump >= 0 && !(W_IsIWADLump(patch_lump) && bigfont_lump >= 0))
+    if (patch_lump >= 0)
     {
         V_DrawPatch(x, y, W_CacheLumpNum(patch_lump, PU_CACHE));
     }
```

Reply:
Maybe we should pass the corresponding `menu_t` over to
`MN_DrawTitle()` and draw the title with the hud font or big font as well if `-> lumps_missing > 0`.

Reply:
> Maybe we should pass the corresponding `menu_t` over to
`MN_DrawTitle()` and draw the title with the hud font or big font as well if `-> lumps_missing > 0`.

Most modern PWADs don't bother with MBF options patches, but still have `M_OPTTTL` (options title). I think we may break some other PWADs with this change.

Of the PWADs I know of with `DBIGFONT`, only Eviternity I-II are missing some menu patches.

Reply:
> Most modern PWADs don't bother with MBF options patches, but still have `M_OPTTTL` (options title). I think we may break some other PWADs with this change.

Well, my idea was to make sure that menu titles are not drawn as patches if the menu items are drawn as text. This would fix Eviternity, but would break e.g. IWADs.

Reply:
> Well, my idea was to make sure that menu titles are not drawn as patches if the menu items are drawn as text. This would fix Eviternity, but would break e.g. IWADs.

I think the current version would work correctly in most cases. It seems trying to fix the Eviternity menu in code was a bad idea.

Reply:
> Well, my idea was to make sure that menu titles are not drawn as patches if the menu items are drawn as text. This would fix Eviternity, but would break e.g. IWADs.

Like this, for example:

```diff
diff --git a/src/mn_menu.c b/src/mn_menu.c
index bd4804b5..df035392 100644
--- a/src/mn_menu.c
+++ b/src/mn_menu.c
@@ -1877,7 +1877,7 @@ void MN_SetNextMenuAlt(ss_types type)
 
 static void M_DrawSetup(void)
 {
-    MN_DrawTitle(108, 15, "M_OPTTTL", "OPTIONS");
+    MN_DrawTitle(108, 15, SetupDef.lumps_missing == -1 ? "M_OPTTTL" : NULL, "OPTIONS");
 }
 
 /////////////////////////////
diff --git a/src/mn_setup.c b/src/mn_setup.c
index d1f106e2..945c4955 100644
--- a/src/mn_setup.c
+++ b/src/mn_setup.c
@@ -3721,7 +3721,12 @@ int MN_StringHeight(const char *string)
 
 void MN_DrawTitle(int x, int y, const char *patch, const char *alttext)
 {
-    int patch_lump = W_CheckNumForName(patch);
+    int patch_lump = -1;
+
+    if (patch)
+    {
+        patch_lump = W_CheckNumForName(patch);
+    }
 
     if (patch_lump >= 0)
     {
```